cherab/tools/observers/group/targettedpixel.py

```diff
@@ -0,0 +1,137 @@
+
+        for observer in value:
+            if not isinstance(observer, TargettedPixel):
+                raise TypeError("The observers attribute of TargettedPixelGroup must be a list or tuple of "
+                                "TargettedPixel. Value {} is not a TargettedPixel.".format(observer))
+
+        # Prevent external changes being made to this list
+        for observer in value:
+            observer.parent = self
+
+        self._observers = tuple(value)
+
+    def add_observer(self, observer):
+        """
+        Adds new line of sight to the group.
+
+        :param TargettedPixel observer: Targetted pixel to add.
+        """
+        if not isinstance(observer, TargettedPixel):
+            raise TypeError("The observer argument must be of type TargettedPixel.")
+        observer.parent = self
+        self._observers = self._observers + (observer,)
+
+    @property
+    def x_width(self):
+        return [pixel.x_width for pixel in self._observers]
+
+    @x_width.setter
+    def x_width(self, value):
+        if isinstance(value, (list, tuple, ndarray)):
+            if len(value) == len(self._observers):
+                for pixel, v in zip(self._observers, value):
+                    pixel.x_width = v
+            else:
+                raise ValueError("The length of 'x_width' ({}) "
+                                 "mismatches the number of pixels ({}).".format(len(value), len(self._sight_lines)))
+        else:
+            for pixel in self._observers:
+                pixel.x_width = value
+
+    @property
+    def y_width(self):
+        return [pixel.y_width for pixel in self._observers]
+
+    @y_width.setter
+    def y_width(self, value):
+        if isinstance(value, (list, tuple, ndarray)):
+            if len(value) == len(self._observers):
+                for pixel, v in zip(self._observers, value):
+                    pixel.y_width = v
+            else:
+                raise ValueError("The length of 'y_width' ({}) "
+                                 "mismatches the number of pixels ({}).".format(len(value), len(self._sight_lines)))
+        else:
+            for pixel in self._observers:
+                pixel.y_width = value
+
+    @property
+    def targets(self):
+        """Targets for preferential sampling"""
+        return [pixel.target for pixel in self._observers]
+
+    @targets.setter
+    def targets(self, value):
+        if isinstance(value, (list, tuple)):
+            if len(value) == len(self._observers):
+                for pixel, v in zip(self._observers, value):
+                    pixel.target = v
+                    v.parent = self
+            else:
+                raise ValueError("The length of 'value' ({}) "
+                                 "mismatches the number of pixels ({}).".format(len(value), len(self._sight_lines)))
+        else:
+            value.parent = self
+            for pixel in self._observers:
+                pixel.target = value
```

Comment:
The correct property name is `pixel.targets`, not `pixel.target`. `TargettedPixel` receives a list of targets, not a single target, so here we have to check if the supplied list contains lists/tuples or primitives. The target primitives must be connected to the same scene graph as the `TargettedPixel`, which means they must have the same root, but changing the parent of the target will affect coordinate space transformations, so the parent should not be changed. I propose the following implementation: 
```python
    @targets.setter
    def targets(self, value):
        if all(isinstance(v, (list, tuple)) for v in value):
            if len(value) == len(self._observers):
                for pixel, v in zip(self._observers, value):
                    pixel.targets = v
            else:
                raise ValueError("The length of 'value' ({}) "
                                 "mismatches the number of pixels ({}).".format(len(value), len(self._observers)))
        else:
            # assuming a list of primitives, the pixel's setter will throw an error if not
            for pixel in self._observers:
                pixel.targets = value
```